radio/bluetooth/bluetooth.go

```diff
@@ -76,10 +79,66 @@ func Disconnect(client ble.Client) error {
 	return nil
 }
 
-func Scan(id string, timeout time.Duration) (map[string]bool, error) {
+func WriteChar(client ble.Client, char *ble.Characteristic, value []byte, noRsp bool) error {
+	err := make(chan error)
+	go func() {
+		err <- client.WriteCharacteristic(char, value, noRsp)
+	}()
+
+	for {
```

Comment:
You have raised a good point here - looking into it further `select` is blocking so the `for` loop wrapped around only runs once and is unnecessary. 
https://gobyexample.com/timeouts
http://stackoverflow.com/questions/37021194/how-are-golang-select-statements-implemented